homeassistant/components/motionblinds_ble/cover.py

```diff
@@ -99,21 +95,13 @@ def __init__(self, device: MotionDevice, entry: ConfigEntry) -> None:
             entry.data[CONF_BLIND_TYPE],
             BLIND_TYPE_TO_CLASS[entry.data[CONF_BLIND_TYPE]].__name__,
         )
-        super().__init__()
-        self._device = device
+        MotionblindsBLEEntity.__init__(self, device, entry)
+        CoverEntity.__init__(self)
+        self._attr_unique_id: str = entry.data[CONF_ADDRESS]
+        self.entity_description = COVER_TYPES[entry.data[CONF_BLIND_TYPE]]
         self._device.register_running_callback(self.async_update_running)
         self._device.register_position_callback(self.async_update_position)
         self._device.register_connection_callback(self.async_update_connection)
```

Comment:
Maybe also move this to the base class, together with the async_update_connection function.
I am guessing all entities will need to write their state if it connects.
If more is needed a specific entity can always overwrite the async_update_connection  function.

Reply:
Not all entities need a connection callback, running callback or position callback. The connection callback is only used here to call async_write_ha_state which will refresh the extra state attribute